docs/nginx/events.md

```diff
@@ -0,0 +1,54 @@
+Semantic conventions for Nginx Ingress Controller events
+
+**Status**: [Experimental](../../../document-status.md)
+
+<details>
+<summary>Table of Contents</summary>
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [Access](#access)
+- [Error](#error)
+
+<!-- tocstop -->
+
+</details>
+
+This document describes the semantic conventions for events in Nginx Ingress Controller. All Nginx Ingress Controller events MUST use a namespace of `nginx_ingress_controller` in the `event.name` property.
+
+## Access
+
+The event name MUST be `nginx_ingress_controller.access`.
+
+This event describes the details of an the Nginx access.
+
+| Body Field | Type | Description | Examples | Requirement Level |
+|---|---|---|---|---|
+| `http.request.id` | string | The randomly generated ID of the request | `529a007902362a5f51385a5fa7049884` | Required |
+| `http.request.length` | int | The request length (including request line, header, and request body) | `89` | Recommended |
+| `http.request.time` | double | Time elapsed since the first bytes were read from the client | `0.001`| Recommended |
+| `remote_ip_list` | array | An array of remote IP addresses. It is a list because it is common to include, besides the client IP address, IP addresses from headers like X-Forwarded-For. Real source IP is restored to source.ip. | `["192.168.64.1"]` | Recommended |
+| `upstream.alternative_name` | string | The name of the alternative upstream. |  | Recommended |
+| `upstream.ip` | string | The IP address of the upstream server. If several servers were contacted during request processing, their addresses are separated by commas. | `172.17.0.5` |Recommended |
```

Comment:
In my brain things are opposite: `upstream.*` is the HTTP client and `downstream.*` is the service that gets called to process the request and returns the response back to nginx.

I did google and find https://en.wikipedia.org/wiki/Upstream_server though which matches the definition here.

Given what appears to be [widespread ambiguity](https://stackoverflow.com/a/59458012), could we avoid the term "upstream" somehow?

Maybe something like:

`client.*` = HTTP client
`server.*` = nginx server itself
`backend.*` = (what's referred to as "upstream" in this PR)

I guess it depends though on whether the goal is to define a (mostly) generic access log event, or to define a very specific nginx access log, in which case it could make more sense to stick with whatever terms / fields names nginx uses.